### DISCUSSION TITLE
Get relation name instead of relation id in authModel

### DISCUSSION THREAD

**Discussion by starredev** [pocketbase/pocketbase#1807] [Q&A]
Is it possible to get relation name instead of relation id in authModel?

I tried serveral things without luck

**Comment by ganigeorgiev** [comment]
The record response will always return only the id of the relations (aka. the same representation as in the db).

If you want to get the relation fields, you can use [expand](https://pocketbase.io/docs/expanding-relations/) (it can be passed even to the auth methods). Here is an example using the JS SDK:

```js
await pb.collection("users").authWithPassword("test@example.com", "1234567890", {}, {
  expand: "someRelationField1,someRelationField2.subRel"
})

console.log(pb.authStore.model.expand)
```

If that's not what you are looking for, could you elaborate a little more on your use case (sample/pseudo code would be also helpful)?

**Reply by wjoseph0** [reply]
My code properly expanded the relation fields after adding the extra `{},`. I didn't have to add this for List/Search and View. Why is the extra `{},` necessary here?

**Reply by ganigeorgiev** [reply]
@wjoseph0 Because `expand` is a query parameter. The first `{}` in the above refer to the optional body parameters that could be sent with the request. `getList` and `getOne` methods are GET requests and they don't have body. Here is the definition of the methods for clarity:

```js
pb.collection(collectionIdOrName).getList(page = 1, perPage = 30, queryParams = {});

pb.collection(collectionIdOrName).getFullList(batch = 200, queryParams = {});

pb.collection(collectionIdOrName).getFirstListItem(filter, queryParams = {});

pb.collection(collectionIdOrName).getOne(recordId, queryParams = {});

---

pb.collection(collectionIdOrName).authWithPassword(usernameOrEmail, password, bodyParams = {}, queryParams = {});

pb.collection(collectionIdOrName).authRefresh(bodyParams = {}, queryParams = {});

```

(_you can find a list with all available methods in https://github.com/pocketbase/js-sdk#services_)

**Reply by wjoseph0** [reply]
Thank you I understand now

**Comment by starredev** [comment]
@ganigeorgiev when I refresh the page while I'm logged in it seems that pb.authStore.model.expand lost the relation. 

Any idea? or is that how it is designed

**Reply by ganigeorgiev** [reply]
If you are calling `authRefresh()`, you'll have to specify the `expand` param there too:

```js
await pb.collection("users").authRefresh({}, {
  expand: "someRelationField1,someRelationField2.subRel"
})
```

**Reply by starredev** [reply]
I'm not using AuthRefresh, but i do it other way thx

**Comment by starredev** [comment]
@ganigeorgiev there must be an bug somewhere.

When I look in the db.authStore object I see expand is showing up there.
When I look in to the db.authStore.model the expand is empty until page reload

**Reply by ganigeorgiev** [reply]
I'm not sure that I understand what is the issue.

What is `db` in the above - is this the PocketBase instance and if yes why `authStore.token` is not prefixed or it is something custom?
Where your code is executed? Is this in a client-side or server-side (SSR) context?

The following works as expected for me (including after page reload and commenting the `authWithPassword` part):

```js
await pb.collection('users').authWithPassword('test@example.com', '1234567890', {}, {
    expand: 'someRelField'
})
...
console.log(pb.authStore)
console.log(pb.authStore.model)
```

**Reply by starredev** [reply]
I got it worked! thanks anyway :)

**Comment by starredev** [comment]
Close please, i got it working.
The patch i did in pinia was killing it
